web/srv/api_handlers.go

```diff
@@ -168,10 +170,34 @@ func (h *handler) handleAPITopRoutes(w http.ResponseWriter, req *http.Request, p
 	renderJSONPb(w, result)
 }
 
+// Control frame payload size must be no longer than 123 bytes. In the case of
+// an unexpected HTTP status code or unexpected error, truncate the message
+// after 123 bytes so that the web socket message is properly written.
+func createTapErrorMessage(err error) string {
+	var msg string
+
+	if httpErr, ok := err.(protohttp.HTTPError); ok {
+		log.Debugf("tap error is HTTPError: %s", httpErr.Error())
+		if httpErr.Code == http.StatusForbidden {
+			msg = fmt.Sprintf("tap authorization failed, visit %s for more information", tap.TapRbacURL)
+		} else {
+			msg = httpErr.WrappedError.Error()[:123]
+		}
+	} else {
+		log.Debugf("tap error: %s", err.Error())
+		msg = err.Error()[:123]
+	}
+
+	return msg
+}
+
 func websocketError(ws *websocket.Conn, wsError int, msg string) {
-	ws.WriteControl(websocket.CloseMessage,
+	err := ws.WriteControl(websocket.CloseMessage,
```

Comment:
i'd modify `websocketError` to take an `err error` rather than `msg string`, and then call `createTapErrorMessage` from within this function, otherwise other calls to `websocketError` still could hit that character limit.